squeal-postgresql/src/Squeal/PostgreSQL/Binary.hs

```diff
@@ -594,6 +600,14 @@ instance FromValue 'PGtext Strict.Text where fromValue = Decoding.text_strict
 instance FromValue 'PGtext Lazy.Text where fromValue = Decoding.text_lazy
 instance FromValue 'PGtext String where
   fromValue = Strict.Text.unpack <$> Decoding.text_strict
+instance KnownNat n => FromValue ('PGvarchar n) (VarChar n) where
+  fromValue = (constructVarChar <$> Decoding.text_strict) >>= \case
+      Just t -> return t
+      Nothing -> failure "Could not construct VarChar."
+instance KnownNat n => FromValue ('PGchar n) (VarChar n) where
```

Comment:
I'm not sure this is right. `CHAR(n)` are fixed length, you'd need another newtype to describe them.

Reply:
As far as I know, `char` would just get padded with spaces if the length is smaller.

It depends on whether we actually want that.

Reply:
I think it would actually be better to force the exact length.